new play youtube-clone created

Srinu126 opened the pull request:
# Description
#Youtube Clone
  I have created a simple Video viewer app similar to Youtube. I am making use of google api to fetch the videos. The user can search for the videos and once he submit the search key, the videos will be fetched and rendered on the UI.
Fixes # (issue)
This is a new Play and it is not breaking any existing code. And also, I believe there is still some css changes to be made and I will be working on that.

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

Srinu126 commented:
@atapas could you please review and approve the pull request. Thanks !

atapas commented:
@Srinu126 please create an issue for this PR and add in the PR description. Also, please read through the check list for your reference: https://github.com/atapas/react-play/wiki/ReactPlay-Code-Review-Checklist

Srinu126 commented:
> @Srinu126 please create an issue for this PR and add in the PR description. Also, please read through the check list for your reference: https://github.com/atapas/react-play/wiki/ReactPlay-Code-Review-Checklist

This is the first time I am raising a PR to external Repo. I am not sure what you are talking about. could you please guide me here

atapas commented:
> > @Srinu126 please create an issue for this PR and add in the PR description. Also, please read through the check list for your reference: https://github.com/atapas/react-play/wiki/ReactPlay-Code-Review-Checklist
> 
> This is the first time I am raising a PR to external Repo. I am not sure what you are talking about. could you please guide me here

No problem.

Give me a title and description of the work you are doing. I will do the rest(for the first time) and let you know so that you can do it from the next time.

atapas commented:
> @Srinu126

Hi @Srinu126 

I have added the issue: https://github.com/atapas/react-play/issues/190

Please check if all the details are added correctly.

atapas commented on src/plays/youtube-clone/Readme.md:
```diff
@@ -0,0 +1 @@
+# Youtube-Clone
```
Please provide details in your Readme file. You can refer to the readme file from other plays as a reference.

atapas commented on src/plays/youtube-clone/SearchBar.jsx:
```diff
@@ -0,0 +1,20 @@
+            <div className='ui segment'>
+                <form onSubmit={this.onSubmit} className='ui form'>
+                    <label>Video Search</label>
+                    <input style={{width:'350px',marginLeft:'10px',borderRadius:'5px'}} type="text" value={this.state.searchValue} onChange={(event)=>this.setState({searchValue: event.target.value})} />
```
Avoid inline styles.

atapas commented on src/plays/youtube-clone/youtube.js:
```diff
@@ -0,0 +1,12 @@
+import axios from 'axios';
+const KEY = 'AIzaSyCO5HO9G8OyvzdnQn8DGSkfwzQIrXkYo0s'
```
Do not expose the keys. It may bring security issues to your account.

Please create a file called `.env` at the root of the project folder and keep the key there instead. You can read more about it from here: https://create-react-app.dev/docs/adding-custom-environment-variables/

atapas commented on src/plays/youtube-clone/VideoViewer.jsx:
```diff
@@ -0,0 +1,41 @@
+                <SearchBar handleSearch={this.handleSearchValue} />
+                <div className='ui grid'>
+                    <div style={{display:'flex'}} className='ui row'>
+                        <div style={{height:'500px',marginTop:'10px'}} className='eleven wide column'>
```
No inline style

atapas commented on src/plays/youtube-clone/VideoViewer.jsx:
```diff
@@ -0,0 +1,41 @@
+                        <div style={{height:'500px',marginTop:'10px'}} className='eleven wide column'>
+                            {this.state.selectedVideo ? <VideoDetail video={this.state.selectedVideo} /> : null}
+                        </div>
+                        <div className='five wide column'>
```
Where do you use these class names in your stye? Also, are you sure to use 3 class selectors here?

Srinu126 replied:
Actually I am using semantic UI cdn. that is the reason I gave those classnames

atapas replied:
Can we use the style written by us than semantic CSS? It is because We have decided to use the following style system support,

- SASS
- Raw CSS
- TailwindCSS(Upcoming)

I wonder if the entire app is not going for the semantic CSS, there are higher chances of overriding. I think, for this app, it may be easy to just write or own CSS or SASS.

atapas reviewed (changes requested):
@Srinu126 Thank you for submitting the PR. I have done an initial code review and provided my comments. I would suggest you look into the Code Checklist and readjust the code accordingly wherever needed.

Here is the link: https://github.com/atapas/react-play/wiki/ReactPlay-Code-Review-Checklist

Also, I have tried the play with the preview link: https://react-play-git-fork-srinu126-main-atapas.vercel.app/plays/youtube-clone

Here are some comments related to the UI/UX

- The Search and Search result is far apart. I think, you need to bring the video list under the search bar and the details at the right.
![image](https://user-images.githubusercontent.com/3633137/168950177-f5f302a0-d42a-4b16-9083-6535716bae6c.png)
- Clicking on the details doesn't show the video preview. Also, the texts are not formatted.
![image](https://user-images.githubusercontent.com/3633137/168950247-947a23d5-39ff-427e-b00c-5e4fcb6a6146.png)
- The styling of the list can be made better. There is no space between them
![image](https://user-images.githubusercontent.com/3633137/168950299-647de3bf-2eb4-4227-83c8-a4f12f0df0e4.png)
- The UI doesn't work well in the mobile/responsive mode. You have to use media queries to tackle them.
![image](https://user-images.githubusercontent.com/3633137/168950401-ac3d1978-c575-487c-8501-030658626822.png)

I think it is still a work in progress as it doesn't look complete. You can take care of the comments and let us know when ready. We will review it back.

Thanks for your effort. Keep making things better and you will learn a lot. All the best.

Srinu126 commented:
I have fixed the issues like keeping the api key in .env file, removed inline styling and also made SearchBar Component a functional component and also made the video preview available. everything is working fine and the only thing i need to work on is responsiveness and I will be working on it.

atapas commented:
@Srinu126 Whatever comments are taken care, could you please mark them as resolved by clicking the Resolve Conversation button? That will help me to manage the pending comments better.

atapas commented on src/plays/youtube-clone/VideoViewer.jsx:
```diff
@@ -0,0 +1,41 @@
+import youtube from './youtube';
+import VideoList from './VideoList';
+import VideoDetail from './VideoDetail';
+class VideoViewer extends React.Component {
```
All the class component change to function.

atapas commented:
@Srinu126 Please drop a "Review Ready" note when you want us to pick up for a review. Thanks!

atapas commented:
Hello @Srinu126 

We will be moving the react-play project from the `atapas` account to an organization to help manage the project in a better way. The new organization URL is: https://github.com/reactplay.

> **The movement will take place between 12:30 - 14:00 hrs IST(7:00 am - 8:30 am GMT), 30th May 2022. Please refrain from pushing any changes between this time.**

We will inform you when the movement is complete. You need to make a few minimal configuration changes after that. We will let you know.

atapas commented:
> Hello @Srinu126
> 
> We will be moving the react-play project from the `atapas` account to an organization to help manage the project in a better way. The new organization URL is: https://github.com/reactplay.
> 
> > **The movement will take place between 12:30 - 14:00 hrs IST(7:00 am - 8:30 am GMT), 30th May 2022. Please refrain from pushing any changes between this time.**
> 
> We will inform you when the movement is complete. You need to make a few minimal configuration changes after that. We will let you know.

The migration is complete. You can find the project here: https://github.com/reactplay

You no need to reclone the project. You just need to set the correct upstream. Here is the link: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/configuring-a-remote-for-a-fork

atapas commented:
Hey @Srinu126 

A reminder to get back on this play. This PR may be closed due to a lack of activities soon. Please let us know if you need any help.

Looking forward to getting the play in!

atapas commented:
Closing as NO RESPONE.
